frontend/ui/elements/screensaver_menu.lua

```diff
@@ -82,19 +80,18 @@ return {
             {
                 text = _("Custom images"),
                 enabled_func = function()
-                    return G_reader_settings:readSetting("screensaver_type") == "image_file"
-                            or G_reader_settings:readSetting("screensaver_type") == "random_image"
-                            or G_reader_settings:readSetting("screensaver_type") == "document_cover"
+                    return G_reader_settings:readSetting("screensaver_type") == "random_image"
+                           or G_reader_settings:readSetting("screensaver_type") == "document_cover"
                 end,
                 sub_item_table = {
                     {
-                        text = _("Select custom image"),
+                        text = _("Select image or document cover"),
                         enabled_func = function()
-                            return G_reader_settings:readSetting("screensaver_type") == "image_file"
+                            return G_reader_settings:readSetting("screensaver_type") == "document_cover"
                         end,
                         keep_menu_open = true,
                         callback = function()
-                            Screensaver:chooseFile()
+                            Screensaver:chooseFile(true)
```

Comment:
So, we're only calling chooseFile() with true, which (as it seems you have not modified it) will only allow selecting documents, not cover.
Or is it that all this merging is just because an image is actually a document :) so we can just choose an image as a document ? (didn't think about it like that until now...)

Reply:
Full details in https://github.com/koreader/koreader/pull/11549/commits/cf4fe98146d459f61ea7f60778fa5d680c07d473

I'm detecting image files and sending those off to ImageWidget (which has some logic to render e.g. SVG at the right resolution).

I'd overlooked the chooseFile thing. I'll remove that.

Reply:
And caught this stray message in the process.
<img src=https://github.com/koreader/koreader/assets/202757/171b3a7c-a48b-4098-ab4b-16d3ce062b9c width=45%>

Reply:
Why strange? It's an intermediate screen showing the current value, and allowing you to go choosing another one.

Reply:
Stray as in it still said `Current screensaver document cover:` :-)